packages/components/src/systems/PatientMedHistory/index.tsx

```diff
@@ -0,0 +1,185 @@
+import { createSignal, createEffect, Show, For, createMemo } from 'solid-js';
+import gql from 'graphql-tag';
+import { usePhotonClient } from '../SDKProvider';
+import { PhotonClient } from '@photonhealth/sdk';
+import { PatientMedication, Medication, SearchMedication } from '@photonhealth/sdk/dist/types';
+import Table from '../../particles/Table';
+import Text from '../../particles/Text';
+import generateString from '../../utils/generateString';
+import Badge from '../../particles/Badge';
+import formatDate from '../../utils/formatDate';
+import Button from '../../particles/Button';
+
+const GET_PATIENT_MED_HISTORY = gql`
+  query GetPatient($id: ID!) {
+    patient(id: $id) {
+      id
+      medicationHistory {
+        active
+        prescription {
+          id
+          writtenAt
+        }
+        medication {
+          id
+          name
+        }
+      }
+    }
+  }
+`;
+
+const ADD_MED_HISTORY = gql`
+  mutation UpdateMedicationHistory($id: ID!, $medicationHistory: [MedHistoryInput!]!) {
+    updatePatient(id: $id, medicationHistory: $medicationHistory) {
+      id
+    }
+  }
+`;
+
+type PatientMedHistoryProps = {
+  patientId: string;
+  openAddMedication?: () => void;
+  newMedication?: Medication | SearchMedication;
+};
+
+const LoadingRowFallback = () => (
+  <Table.Row>
+    <Table.Cell>
+      <Text sampleLoadingText={generateString(10, 45)} loading />
+    </Table.Cell>
+    <Table.Cell>
+      <Text sampleLoadingText={generateString(6, 8)} loading />
+    </Table.Cell>
+    <Table.Cell>
+      <Text sampleLoadingText={generateString()} loading />
+    </Table.Cell>
+    <Table.Cell>
+      <Text sampleLoadingText={generateString()} loading />
+    </Table.Cell>
+  </Table.Row>
+);
+
+export default function PatientMedHistory(props: PatientMedHistoryProps) {
+  const client = usePhotonClient();
+  const [medHistory, setMedHistory] = createSignal<PatientMedication[] | undefined>(undefined);
+
+  const baseURL = createMemo(() => `${client?.clinicalUrl}prescriptions/`);
+
+  const fetchPatient = async () => {
+    const { data } = await client!.apollo.query({
+      query: GET_PATIENT_MED_HISTORY,
+      variables: { id: props.patientId }
+    });
+    if (data?.patient?.medicationHistory) {
+      setMedHistory(data.patient.medicationHistory);
+    }
+  };
+
+  const addMedHistory = async (medicationId: string) => {
+    await (client as PhotonClient)!.apollo.mutate({
```

Comment:
Why does this need the type assertion?

Reply:
this one is a true murder mystery, IDE was acting up before, but now I just removed and it's fine :|